extensions/pkg/controller/worker/genericactuator/machine_controller_manager.go

```diff
@@ -124,8 +125,58 @@ func (a *genericActuator) waitUntilMachineControllerManagerIsDeleted(ctx context
 }
 
 func (a *genericActuator) scaleMachineControllerManager(ctx context.Context, logger logr.Logger, worker *extensionsv1alpha1.Worker, replicas int32) error {
+	deployment := &appsv1.Deployment{}
+	if err := a.client.Get(ctx, kutil.Key(worker.Namespace, McmDeploymentName), deployment); err != nil {
+		if apierrors.IsNotFound(err) {
+			logger.Info("Machine-controller-manager is not deployed yet")
+			return nil
+		}
+		return err
+	}
 	logger.Info("Scaling machine-controller-manager", "replicas", replicas)
-	return kubernetes.ScaleDeployment(ctx, a.client, kutil.Key(worker.Namespace, McmDeploymentName), replicas)
+	if err := kubernetes.ScaleDeployment(ctx, a.client, kutil.Key(worker.Namespace, McmDeploymentName), replicas); err != nil {
+		return err
+	}
```

Comment:
Could you use https://github.com/gardener/gardener/blob/03794c4fab012f16625540cd2989ab8f328be93d/pkg/client/kubernetes/scaling.go#L54-L63?

Reply:
I think this is the same method that it is used